modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/util/MulExtensionUtilsTestCase.java

```diff
@@ -65,6 +80,144 @@ public void optionalWithDefaultValue() {
+    List<ParameterGroupModel> groups = singletonList(parameterGroupModel);
+
+    Map<String, StaticValueResolver> parameters = new HashMap<>();
+    parameters.put("repeatedNameParameter", mock(StaticValueResolver.class));
+    Map<String, String> aliasedParameterNames = new HashMap<>(0);
+    checkParameterGroupExclusiveness(Optional.of(mock(OperationModel.class)), groups, parameters, aliasedParameterNames);
+  }
+
+  @Test
+  public void parameterGroupExclusivenessForDslFalseComplexParameter() throws ConfigurationException {
+    ExclusiveParametersModel exclusiveParametersModel = mock(ExclusiveParametersModel.class);
+    when(exclusiveParametersModel.getExclusiveParameterNames())
+        .thenReturn(new HashSet<>(asList("someParameter", "repeatedNameParameter", "complexParameter")));
+
+    ParameterGroupModel parameterGroupModel = mock(ParameterGroupModel.class);
+    when(parameterGroupModel.getExclusiveParametersModels()).thenReturn(singletonList(exclusiveParametersModel));
+    when(parameterGroupModel.isShowInDsl()).thenReturn(false);
+    List<ParameterGroupModel> groups = singletonList(parameterGroupModel);
+
+    Map<String, Object> parameters = new HashMap<>();
+    parameters.put("complexParameter", mock(ObjectBuilderValueResolver.class));
+    Map<String, String> aliasedParameterNames = new HashMap<>(0);
+    checkParameterGroupExclusiveness(Optional.of(mock(OperationModel.class)), groups, parameters, aliasedParameterNames);
+  }
+
+  @Test
+  public void parameterGroupExclusivenessForDslFalseComplexParameterDynamic()
+      throws ConfigurationException, ValueResolvingException {
+    ExclusiveParametersModel exclusiveParametersModel = mock(ExclusiveParametersModel.class);
+    when(exclusiveParametersModel.getExclusiveParameterNames())
+        .thenReturn(new HashSet<>(asList("someParameter", "repeatedNameParameter", "complexParameter")));
+
+    ParameterGroupModel parameterGroupModel = mock(ParameterGroupModel.class);
+    when(parameterGroupModel.getExclusiveParametersModels()).thenReturn(singletonList(exclusiveParametersModel));
+    when(parameterGroupModel.isShowInDsl()).thenReturn(false);
+    List<ParameterGroupModel> groups = singletonList(parameterGroupModel);
+
+    Map<String, ValueResolver<?>> parameterResolvers = new HashMap<>();
+    parameterResolvers.put("anotherParameter", mock(StaticValueResolver.class));
+    parameterResolvers.put("repeatedNameParameter", mock(StaticValueResolver.class));
+    ObjectBuilderValueResolver valueResolver = mock(ObjectBuilderValueResolver.class);
+    when(valueResolver.getParameters()).thenReturn(parameterResolvers);
+    Map<String, ObjectBuilderValueResolver> parameters = new HashMap<>();
+    parameters.put("complexParameter", valueResolver);
+    Map<String, String> aliasedParameterNames = new HashMap<>(0);
+    checkParameterGroupExclusiveness(Optional.of(mock(OperationModel.class)), groups, parameters, aliasedParameterNames);
+  }
+
```

Comment:
Add a test where there is another parameter group, that is not show in DSL, that has a parameter that is a POJO, and has a field with the name of a parameter in the exclusive optional parameter group that is Show in DSL.